rules_gapic/gapic.bzl

```diff
@@ -12,37 +12,38 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+def _set_args(arg, arg_name, args, inputs = None):
+    if not arg:
+        return
+    args.append("%s%s" % (arg_name, arg.files.to_list()[0].path if hasattr(arg, "files") else arg))
+    if inputs != None:
+        inputs.append(arg.files.to_list()[0])
+
 def _gapic_srcjar_impl(ctx):
-    output = ctx.outputs.output
-    optional_arguments = []
-    optional_action_inputs = []
-
-    if ctx.file.service_yaml:
-        optional_arguments.append("--service_yaml=%s" % ctx.file.service_yaml.path)
-        optional_action_inputs.append(ctx.file.service_yaml)
-    if ctx.attr.package_yaml2:
-        optional_arguments.append("--package_yaml2=%s" % ctx.file.package_yaml2.path)
-        optional_action_inputs.append(ctx.file.package_yaml2)
-
-    if ctx.attr.artifact_type.find("DISCOGAPIC") >= 0:
-        optional_arguments.append("--discovery_doc=%s" % ctx.file.src.path)
+    arguments = []
+    inputs = []
+
+    attr = ctx.attr
+    if attr.artifact_type:
+        _set_args(attr.artifact_type, "", arguments)
+        if ctx.attr.artifact_type.find("DISCOGAPIC") >= 0:
+            _set_args(attr.src, "--discovery_doc=", arguments, inputs)
+        else:
```

Comment:
`service_yaml` will go away soon as it is expected to be fully freplaced with new annotations. Removing it here (maybe a little bit to early, but it feels like an ok thing to do). WDYT?

Reply:
I think we could this validation logic to the generator to ensure that if a GAPIC v1 is given, that a service yaml is also given. Feel free to do that in this PR or file an issue.

Reply:
The service yaml can't be fully replaced by the annotations because of LRO http bindings. Annotations for those aren't even in the works. It will still be mandatory in gapic-generator for a while.

Reply:
@eoogbe but LRO is needed only by a subset of clients. Note, if at least one client can be generated without service.yaml, it becomes optional.